Compiling and running base_code.cpp for Task 303997...
In file included from /usr/local/include/gtest/gtest.h:71,
                 from tasks/303997/test.cpp:1:
tasks/303997/test.cpp: In member function 'virtual void AddTest_ArithmeticOperations_Test::TestBody()':
tasks/303997/test.cpp:33:25: error: call of overloaded 'add(int, double)' is ambiguous
   33 |     EXPECT_DOUBLE_EQ(add(3, 3.14), 6.14);
      |                      ~~~^~~~~~~~~
In file included from tasks/303997/test.cpp:7:
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/test.cpp:34:24: error: call of overloaded 'add(float, int)' is ambiguous
   34 |     EXPECT_FLOAT_EQ(add(2.5f, 2), 4.5f);
      |                     ~~~^~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/test.cpp: In member function 'virtual void AddTest_CharAndIntAsciiAddition_Test::TestBody()':
tasks/303997/test.cpp:49:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second:
   49 |     EXPECT_EQ(add('A', 5), 70);   // 'A' (65) + 5 = 70
      |               ~~~^~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate 1: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:20:13: note: candidate 2: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/test.cpp:50:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second:
   50 |     EXPECT_EQ(add(10, 'B'), 76);  // 'B' (66) + 10 = 76
      |               ~~~^~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate 1: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:20:13: note: candidate 2: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/test.cpp: In member function 'virtual void AddTest_CharStringConcatenation_Test::TestBody()':
tasks/303997/test.cpp:56:18: error: no matching function for call to 'add(char, std::string)'
   56 |     EXPECT_EQ(add('A', std::string("BC")), "ABC");
      |               ~~~^~~~~~~~~~~~~~~~~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:20: note:   no known conversion for argument 2 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'int'
    5 | int add(int a, int b) {
      |                ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:26: note:   no known conversion for argument 2 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'float'
   10 | float add(float a, float b) {
      |                    ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:29: note:   no known conversion for argument 2 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'double'
   15 | double add(double a, double b) {
      |                      ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:30: note:   no known conversion for argument 2 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'char'
   20 | std::string add(char a, char b) {
      |                         ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:36: note:   no known conversion for argument 1 from 'char' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                 ~~~~~~~~~~~~~~~~~~~^
tasks/303997/test.cpp:57:18: error: no matching function for call to 'add(std::string, char)'
   57 |     EXPECT_EQ(add(std::string("Hi"), '!'), "Hi!");
      |               ~~~^~~~~~~~~~~~~~~~~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:13: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'int'
    5 | int add(int a, int b) {
      |         ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:17: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'float'
   10 | float add(float a, float b) {
      |           ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:19: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'double'
   15 | double add(double a, double b) {
      |            ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:22: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'char'
   20 | std::string add(char a, char b) {
      |                 ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:58: note:   no known conversion for argument 2 from 'char' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                                       ~~~~~~~~~~~~~~~~~~~^
tasks/303997/test.cpp: In member function 'virtual void AddTest_UserDefinedComplexNumber_Test::TestBody()':
tasks/303997/test.cpp:71:18: error: no matching function for call to 'add(ComplexNumber&, ComplexNumber&)'
   71 |     EXPECT_EQ(add(c1, c2), expected);
      |               ~~~^~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:13: note:   no known conversion for argument 1 from 'ComplexNumber' to 'int'
    5 | int add(int a, int b) {
      |         ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:17: note:   no known conversion for argument 1 from 'ComplexNumber' to 'float'
   10 | float add(float a, float b) {
      |           ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:19: note:   no known conversion for argument 1 from 'ComplexNumber' to 'double'
   15 | double add(double a, double b) {
      |            ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:22: note:   no known conversion for argument 1 from 'ComplexNumber' to 'char'
   20 | std::string add(char a, char b) {
      |                 ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:36: note:   no known conversion for argument 1 from 'ComplexNumber' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                 ~~~~~~~~~~~~~~~~~~~^
In file included from /usr/local/include/gtest/gtest-printers.h:122,
                 from /usr/local/include/gtest/gtest-matchers.h:49,
                 from /usr/local/include/gtest/internal/gtest-death-test-internal.h:47,
                 from /usr/local/include/gtest/gtest-death-test.h:43,
                 from /usr/local/include/gtest/gtest.h:64:
tasks/303997/test.cpp: In member function 'virtual void AddTest_ThrowsOnUnsupportedOperation_Test::TestBody()':
tasks/303997/test.cpp:76:21: error: no matching function for call to 'add(int, std::vector<int>)'
   76 |     EXPECT_THROW(add(5, std::vector<int>{}), std::invalid_argument);
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:20: note:   no known conversion for argument 2 from 'std::vector<int>' to 'int'
    5 | int add(int a, int b) {
      |                ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:26: note:   no known conversion for argument 2 from 'std::vector<int>' to 'float'
   10 | float add(float a, float b) {
      |                    ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:29: note:   no known conversion for argument 2 from 'std::vector<int>' to 'double'
   15 | double add(double a, double b) {
      |                      ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:30: note:   no known conversion for argument 2 from 'std::vector<int>' to 'char'
   20 | std::string add(char a, char b) {
      |                         ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:36: note:   no known conversion for argument 1 from 'int' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                 ~~~~~~~~~~~~~~~~~~~^
tasks/303997/test.cpp:77:21: error: no matching function for call to 'add(std::string, std::vector<int>)'
   77 |     EXPECT_THROW(add(std::string("Test"), std::vector<int>{1, 2, 3}), std::invalid_argument);
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:13: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'int'
    5 | int add(int a, int b) {
      |         ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:17: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'float'
   10 | float add(float a, float b) {
      |           ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:19: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'double'
   15 | double add(double a, double b) {
      |            ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:22: note:   no known conversion for argument 1 from 'std::string' {aka 'std::__cxx11::basic_string<char>'} to 'char'
   20 | std::string add(char a, char b) {
      |                 ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:58: note:   no known conversion for argument 2 from 'std::vector<int>' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                                       ~~~~~~~~~~~~~~~~~~~^
tasks/303997/test.cpp:78:21: error: no matching function for call to 'add(double, std::vector<double>)'
   78 |     EXPECT_THROW(add(5.5, std::vector<double>{1.1, 2.2}), std::invalid_argument);
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:20: note:   no known conversion for argument 2 from 'std::vector<double>' to 'int'
    5 | int add(int a, int b) {
      |                ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:26: note:   no known conversion for argument 2 from 'std::vector<double>' to 'float'
   10 | float add(float a, float b) {
      |                    ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:29: note:   no known conversion for argument 2 from 'std::vector<double>' to 'double'
   15 | double add(double a, double b) {
      |                      ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:30: note:   no known conversion for argument 2 from 'std::vector<double>' to 'char'
   20 | std::string add(char a, char b) {
      |                         ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:36: note:   no known conversion for argument 1 from 'double' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                 ~~~~~~~~~~~~~~~~~~~^
tasks/303997/test.cpp:79:21: error: no matching function for call to 'add(std::vector<int>, std::vector<int>)'
   79 |     EXPECT_THROW(add(std::vector<int>{1, 2}, std::vector<int>{3, 4}), std::invalid_argument);
      |                  ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tasks/303997/base_code.cpp:5:5: note: candidate: 'int add(int, int)'
    5 | int add(int a, int b) {
      |     ^~~
tasks/303997/base_code.cpp:5:13: note:   no known conversion for argument 1 from 'std::vector<int>' to 'int'
    5 | int add(int a, int b) {
      |         ~~~~^
tasks/303997/base_code.cpp:10:7: note: candidate: 'float add(float, float)'
   10 | float add(float a, float b) {
      |       ^~~
tasks/303997/base_code.cpp:10:17: note:   no known conversion for argument 1 from 'std::vector<int>' to 'float'
   10 | float add(float a, float b) {
      |           ~~~~~~^
tasks/303997/base_code.cpp:15:8: note: candidate: 'double add(double, double)'
   15 | double add(double a, double b) {
      |        ^~~
tasks/303997/base_code.cpp:15:19: note:   no known conversion for argument 1 from 'std::vector<int>' to 'double'
   15 | double add(double a, double b) {
      |            ~~~~~~~^
tasks/303997/base_code.cpp:20:13: note: candidate: 'std::string add(char, char)'
   20 | std::string add(char a, char b) {
      |             ^~~
tasks/303997/base_code.cpp:20:22: note:   no known conversion for argument 1 from 'std::vector<int>' to 'char'
   20 | std::string add(char a, char b) {
      |                 ~~~~~^
tasks/303997/base_code.cpp:27:13: note: candidate: 'std::string add(const std::string&, const std::string&)'
   27 | std::string add(const std::string& a, const std::string& b) {
      |             ^~~
tasks/303997/base_code.cpp:27:36: note:   no known conversion for argument 1 from 'std::vector<int>' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   27 | std::string add(const std::string& a, const std::string& b) {
      |                 ~~~~~~~~~~~~~~~~~~~^
Compilation failed for Task 303997
----------------------
Compiling and running base_code.cpp for Task 304005...
tasks/304005/test.cpp:3:10: error: #include expects "FILENAME" or <FILENAME>
    3 | #include WRAPPER_FILE
      |          ^~~~~~~~~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_PositiveTest_Test::TestBody()':
tasks/304005/test.cpp:7:5: error: 'Stack' was not declared in this scope
    7 |     Stack<int> stack(5);
      |     ^~~~~
tasks/304005/test.cpp:7:11: error: expected primary-expression before 'int'
    7 |     Stack<int> stack(5);
      |           ^~~
tasks/304005/test.cpp:10:5: error: 'stack' was not declared in this scope
   10 |     stack.push(10);
      |     ^~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_NegativeTest_Test::TestBody()':
tasks/304005/test.cpp:29:5: error: 'Stack' was not declared in this scope
   29 |     Stack<int> stack(2);
      |     ^~~~~
tasks/304005/test.cpp:29:11: error: expected primary-expression before 'int'
   29 |     Stack<int> stack(2);
      |           ^~~
tasks/304005/test.cpp:30:5: error: 'stack' was not declared in this scope
   30 |     stack.push(10);
      |     ^~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_CornerTest_Test::TestBody()':
tasks/304005/test.cpp:48:5: error: 'Stack' was not declared in this scope
   48 |     Stack<int> stack(1);
      |     ^~~~~
tasks/304005/test.cpp:48:11: error: expected primary-expression before 'int'
   48 |     Stack<int> stack(1);
      |           ^~~
tasks/304005/test.cpp:50:5: error: 'stack' was not declared in this scope
   50 |     stack.push(1);
      |     ^~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_EdgeTest_Test::TestBody()':
tasks/304005/test.cpp:61:5: error: 'Stack' was not declared in this scope
   61 |     Stack<int> stack(maxCapacity);
      |     ^~~~~
tasks/304005/test.cpp:61:11: error: expected primary-expression before 'int'
   61 |     Stack<int> stack(maxCapacity);
      |           ^~~
tasks/304005/test.cpp:65:9: error: 'stack' was not declared in this scope
   65 |         stack.push(static_cast<int>(i));
      |         ^~~~~
In file included from /usr/local/include/gtest/gtest.h:71,
                 from tasks/304005/test.cpp:1:
tasks/304005/test.cpp:69:15: error: 'stack' was not declared in this scope
   69 |     EXPECT_EQ(stack.top(), static_cast<int>(maxCapacity - 1))
      |               ^~~~~
tasks/304005/test.cpp:73:5: error: 'stack' was not declared in this scope
   73 |     stack.pop();
      |     ^~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_MultiThreadedTest_Test::TestBody()':
tasks/304005/test.cpp:87:5: error: 'Stack' was not declared in this scope
   87 |     Stack<int> stack(10);
      |     ^~~~~
tasks/304005/test.cpp:87:11: error: expected primary-expression before 'int'
   87 |     Stack<int> stack(10);
      |           ^~~
tasks/304005/test.cpp: In lambda function:
tasks/304005/test.cpp:92:40: error: 'Stack' does not name a type
   92 |         auto* localStack = static_cast<Stack<int>*>(arg);
      |                                        ^~~~~
tasks/304005/test.cpp:92:45: error: expected '>' before '<' token
   92 |         auto* localStack = static_cast<Stack<int>*>(arg);
      |                                             ^
tasks/304005/test.cpp:92:45: error: expected '(' before '<' token
   92 |         auto* localStack = static_cast<Stack<int>*>(arg);
      |                                             ^
      |                                             (
tasks/304005/test.cpp:92:45: error: expected primary-expression before '<' token
tasks/304005/test.cpp:92:46: error: expected primary-expression before 'int'
   92 |         auto* localStack = static_cast<Stack<int>*>(arg);
      |                                              ^~~
tasks/304005/test.cpp:92:46: error: expected ')' before 'int'
   92 |         auto* localStack = static_cast<Stack<int>*>(arg);
      |                                              ^~~
      |                                              )
tasks/304005/test.cpp: In member function 'virtual void StackTest_MultiThreadedTest_Test::TestBody()':
tasks/304005/test.cpp:105:58: error: 'stack' was not declared in this scope
  105 |     pthread_create(&t1, nullptr, +testStackConcurrency, &stack);
      |                                                          ^~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_StressTest_Test::TestBody()':
tasks/304005/test.cpp:118:5: error: 'Stack' was not declared in this scope
  118 |     Stack<int> stack(10);
      |     ^~~~~
tasks/304005/test.cpp:118:11: error: expected primary-expression before 'int'
  118 |     Stack<int> stack(10);
      |           ^~~
tasks/304005/test.cpp:123:9: error: 'stack' was not declared in this scope
  123 |         stack.push(i);
      |         ^~~~~
tasks/304005/test.cpp:128:9: error: 'stack' was not declared in this scope
  128 |         stack.pop();
      |         ^~~~~
tasks/304005/test.cpp:133:13: error: 'stack' was not declared in this scope
  133 |     while (!stack.isEmpty()) {
      |             ^~~~~
tasks/304005/test.cpp: In member function 'virtual void StackTest_NullPointerTest_Test::TestBody()':
tasks/304005/test.cpp:145:5: error: 'Stack' was not declared in this scope
  145 |     Stack<int>* stack = nullptr;
      |     ^~~~~
tasks/304005/test.cpp:145:11: error: expected primary-expression before 'int'
  145 |     Stack<int>* stack = nullptr;
      |           ^~~
tasks/304005/test.cpp:148:13: error: 'stack' was not declared in this scope
  148 |         if (stack == nullptr) {
      |             ^~~~~
tasks/304005/test.cpp:151:9: error: 'stack' was not declared in this scope
  151 |         stack->push(10);  // This line will not execute if the pointer is null
      |         ^~~~~
Compilation failed for Task 304005
----------------------
Compiling and running base_code.cpp for Task 304016...
In file included from tasks/304016/test.cpp:3:
tasks/304016/base_code.cpp: In function 'bool solveSudoku(std::vector<std::vector<int> >&)':
tasks/304016/base_code.cpp:26:1: warning: no return statement in function returning non-void [-Wreturn-type]
   26 | }
      | ^
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from SudokuSolverTests
[ RUN      ] SudokuSolverTests.SolvableSudoku
Completed Task 304016
----------------------
Compiling and running base_code.cpp for Task 304019...
In file included from tasks/304019/test.cpp:3:
tasks/304019/base_code.cpp: In member function 'std::pair<int, int> PuzzleRay::findZero(const std::vector<std::vector<int> >&)':
tasks/304019/base_code.cpp:31:1: warning: no return statement in function returning non-void [-Wreturn-type]
   31 | }
      | ^
tasks/304019/base_code.cpp: In member function 'std::vector<std::pair<std::vector<std::vector<int> >, int> > PuzzleRay::getNextStates(const std::vector<std::vector<int> >&)':
tasks/304019/base_code.cpp:41:1: warning: no return statement in function returning non-void [-Wreturn-type]
   41 | }
      | ^
tasks/304019/base_code.cpp: In member function 'std::vector<int> PuzzleRay::slidePuzzle(std::vector<std::vector<int> >)':
tasks/304019/base_code.cpp:50:1: warning: no return statement in function returning non-void [-Wreturn-type]
   50 | }
      | ^
Running main() from /Users/ittehad/Documents/Coding/Python/Turing/Tester_cpp_codebench/tasks/303997/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 10 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 10 tests from PuzzleRayTest
[ RUN      ] PuzzleRayTest.ProvidedExample
Completed Task 304019
----------------------
Compiling and running base_code.cpp for Task 304020...
[==========] Running 9 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 9 tests from FFTTest
[ RUN      ] FFTTest.PositiveFFT
tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 5.7402514854763496, which exceeds 1e-1, where
signal[i].real() evaluates to 6.1229349178414392,
originalSignal[i].real() evaluates to 0.38268343236508978, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 10.606601717798211, which exceeds 1e-1, where
signal[i].real() evaluates to 11.313708498984759,
originalSignal[i].real() evaluates to 0.70710678118654746, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 13.858192987669304, which exceeds 1e-1, where
signal[i].real() evaluates to 14.78207252018059,
originalSignal[i].real() evaluates to 0.92387953251128674, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 15, which exceeds 1e-1, where
signal[i].real() evaluates to 16,
originalSignal[i].real() evaluates to 1, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 13.858192987669305, which exceeds 1e-1, where
signal[i].real() evaluates to 14.782072520180591,
originalSignal[i].real() evaluates to 0.92387953251128674, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 10.606601717798215, which exceeds 1e-1, where
signal[i].real() evaluates to 11.313708498984763,
originalSignal[i].real() evaluates to 0.70710678118654757, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 5.740251485476346, which exceeds 1e-1, where
signal[i].real() evaluates to 6.1229349178414356,
originalSignal[i].real() evaluates to 0.38268343236508989, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 5.7402514854763478, which exceeds 1e-1, where
signal[i].real() evaluates to -6.1229349178414374,
originalSignal[i].real() evaluates to -0.38268343236508967, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 10.606601717798211, which exceeds 1e-1, where
signal[i].real() evaluates to -11.313708498984759,
originalSignal[i].real() evaluates to -0.70710678118654746, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 13.8581929876693, which exceeds 1e-1, where
signal[i].real() evaluates to -14.782072520180586,
originalSignal[i].real() evaluates to -0.92387953251128652, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 15, which exceeds 1e-1, where
signal[i].real() evaluates to -16,
originalSignal[i].real() evaluates to -1, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 13.858192987669302, which exceeds 1e-1, where
signal[i].real() evaluates to -14.782072520180588,
originalSignal[i].real() evaluates to -0.92387953251128663, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 10.606601717798215, which exceeds 1e-1, where
signal[i].real() evaluates to -11.313708498984763,
originalSignal[i].real() evaluates to -0.70710678118654768, and
1e-1 evaluates to 0.10000000000000001.

tasks/304020/test.cpp:24: Failure
The difference between signal[i].real() and originalSignal[i].real() is 5.740251485476354, which exceeds 1e-1, where
signal[i].real() evaluates to -6.1229349178414445,
originalSignal[i].real() evaluates to -0.38268343236509039, and
1e-1 evaluates to 0.10000000000000001.

[  FAILED  ] FFTTest.PositiveFFT (0 ms)
[ RUN      ] FFTTest.EdgeCaseSingleElement
[       OK ] FFTTest.EdgeCaseSingleElement (0 ms)
[ RUN      ] FFTTest.HandleEmptyInputSignal
Completed Task 304020
----------------------
Compiling and running base_code.cpp for Task 304031...
tasks/304031/test.cpp: In member function 'virtual void RedisDataBaseTest::SetUp()':
tasks/304031/test.cpp:22:12: error: 'class RedisDataBase' has no member named 'clear'
   22 |         db.clear();
      |            ^~~~~
tasks/304031/test.cpp: In member function 'virtual void RedisDataBaseTest::TearDown()':
tasks/304031/test.cpp:26:12: error: 'class RedisDataBase' has no member named 'clear'
   26 |         db.clear();
      |            ^~~~~
tasks/304031/test.cpp: In member function 'virtual void RedisDataBaseTest_Clear_Test::TestBody()':
tasks/304031/test.cpp:132:8: error: 'class RedisDataBase' has no member named 'clear'
  132 |     db.clear();
      |        ^~~~~
Compilation failed for Task 304031
----------------------
Compiling and running base_code.cpp for Task 304035...
[==========] Running 10 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 10 tests from BilateralFilterTest
[ RUN      ] BilateralFilterTest.PositiveCase
[       OK ] BilateralFilterTest.PositiveCase (0 ms)
[ RUN      ] BilateralFilterTest.NegativeCase_NullImage
[       OK ] BilateralFilterTest.NegativeCase_NullImage (0 ms)
[ RUN      ] BilateralFilterTest.EdgeCase_SinglePixel
tasks/304035/test.cpp:65: Failure
Expected equality of these values:
  image[0][0]
    Which is: 0
  50

[  FAILED  ] BilateralFilterTest.EdgeCase_SinglePixel (0 ms)
[ RUN      ] BilateralFilterTest.CornerCase_LargeKernel
[       OK ] BilateralFilterTest.CornerCase_LargeKernel (0 ms)
[ RUN      ] BilateralFilterTest.StressCase_LargeImage
[       OK ] BilateralFilterTest.StressCase_LargeImage (410 ms)
[ RUN      ] BilateralFilterTest.HighIntensityVariation
tasks/304035/test.cpp:120: Failure
Expected: (image[0][2]) > (128), actual: 0 vs 128

[  FAILED  ] BilateralFilterTest.HighIntensityVariation (0 ms)
[ RUN      ] BilateralFilterTest.HomogeneousImage
tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

tasks/304035/test.cpp:138: Failure
Expected equality of these values:
  image[i][j]
    Which is: 0
  100

[  FAILED  ] BilateralFilterTest.HomogeneousImage (0 ms)
[ RUN      ] BilateralFilterTest.BoundaryCheck
[       OK ] BilateralFilterTest.BoundaryCheck (0 ms)
[ RUN      ] BilateralFilterTest.VeryLargeSigma
tasks/304035/test.cpp:194: Failure
The difference between image[1][1] and average is 43.555555555555557, which exceeds 10, where
image[1][1] evaluates to 49,
average evaluates to 5.4444444444444446, and
10 evaluates to 10.

[  FAILED  ] BilateralFilterTest.VeryLargeSigma (0 ms)
[ RUN      ] BilateralFilterTest.VerySmallSigma
[       OK ] BilateralFilterTest.VerySmallSigma (0 ms)
[----------] 10 tests from BilateralFilterTest (410 ms total)

[----------] Global test environment tear-down
[==========] 10 tests from 1 test suite ran. (410 ms total)
[  PASSED  ] 6 tests.
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] BilateralFilterTest.EdgeCase_SinglePixel
[  FAILED  ] BilateralFilterTest.HighIntensityVariation
[  FAILED  ] BilateralFilterTest.HomogeneousImage
[  FAILED  ] BilateralFilterTest.VeryLargeSigma

 4 FAILED TESTS
Completed Task 304035
----------------------
Compiling and running base_code.cpp for Task 304044...
Running main() from /Users/ittehad/Documents/Coding/Python/Turing/Tester_cpp_codebench/tasks/303997/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from PermutationFactoryTest
[ RUN      ] PermutationFactoryTest.HandlesTestCases
tasks/304044/test.cpp:95: Failure
Expected equality of these values:
  result[i].size()
    Which is: 2
  expected[i].size()
    Which is: 4
Mismatch in number of operations for test case 4

[  FAILED  ] PermutationFactoryTest.HandlesTestCases (0 ms)
[----------] 1 test from PermutationFactoryTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] PermutationFactoryTest.HandlesTestCases

 1 FAILED TEST
Completed Task 304044
----------------------
Compiling and running base_code.cpp for Task 304048...
tasks/304048/test.cpp: In member function 'virtual void StackTest_MaxSizeTest_Test::TestBody()':
tasks/304048/test.cpp:6:21: error: no matching function for call to 'IntStack::IntStack(int)'
    6 |     IntStack stack(5);
      |                     ^
In file included from tasks/304048/test.cpp:2:
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp: In member function 'virtual void StackTest_PopEmptyStackTest_Test::TestBody()':
tasks/304048/test.cpp:18:21: error: no matching function for call to 'IntStack::IntStack(int)'
   18 |     IntStack stack(5);
      |                     ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp: In member function 'virtual void StackTest_TopEmptyStackTest_Test::TestBody()':
tasks/304048/test.cpp:23:22: error: no matching function for call to 'IntStack::IntStack(int)'
   23 |     IntStack stack(10);
      |                      ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp: In member function 'virtual void StackTest_PositiveStackTest_Test::TestBody()':
tasks/304048/test.cpp:28:21: error: no matching function for call to 'IntStack::IntStack(int)'
   28 |     IntStack stack(5);
      |                     ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp:35:37: error: 'class IntStack' has no member named 'monotonicIncreasing'
   35 |     std::vector<int> result = stack.monotonicIncreasing();
      |                                     ^~~~~~~~~~~~~~~~~~~
tasks/304048/test.cpp: In member function 'virtual void StackTest_NegativeStackTest_Test::TestBody()':
tasks/304048/test.cpp:41:21: error: no matching function for call to 'IntStack::IntStack(int)'
   41 |     IntStack stack(5);
      |                     ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp:48:37: error: 'class IntStack' has no member named 'monotonicIncreasing'
   48 |     std::vector<int> result = stack.monotonicIncreasing();
      |                                     ^~~~~~~~~~~~~~~~~~~
tasks/304048/test.cpp: In member function 'virtual void StackTest_MixedStackTest_Test::TestBody()':
tasks/304048/test.cpp:54:21: error: no matching function for call to 'IntStack::IntStack(int)'
   54 |     IntStack stack(5);
      |                     ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp:61:37: error: 'class IntStack' has no member named 'monotonicIncreasing'
   61 |     std::vector<int> result = stack.monotonicIncreasing();
      |                                     ^~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/gtest/gtest-printers.h:122,
                 from /usr/local/include/gtest/gtest-matchers.h:49,
                 from /usr/local/include/gtest/internal/gtest-death-test-internal.h:47,
                 from /usr/local/include/gtest/gtest-death-test.h:43,
                 from /usr/local/include/gtest/gtest.h:64,
                 from tasks/304048/test.cpp:1:
tasks/304048/test.cpp: In member function 'virtual void StackTest_NegativeMaxSizeTest_Test::TestBody()':
tasks/304048/test.cpp:67:35: error: no matching function for call to 'IntStack::IntStack(int)'
   67 |     EXPECT_THROW(IntStack stack(-1), std::runtime_error);
      |                                   ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp: In member function 'virtual void StackTest_EmptyStackTest_Test::TestBody()':
tasks/304048/test.cpp:71:21: error: no matching function for call to 'IntStack::IntStack(int)'
   71 |     IntStack stack(5);
      |                     ^
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack()'
    1 | class IntStack {
      |       ^~~~~~~~
tasks/304048/base_code.cpp:1:7: note:   candidate expects 0 arguments, 1 provided
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(const IntStack&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'const IntStack&'
tasks/304048/base_code.cpp:1:7: note: candidate: 'IntStack::IntStack(IntStack&&)'
tasks/304048/base_code.cpp:1:7: note:   no known conversion for argument 1 from 'int' to 'IntStack&&'
tasks/304048/test.cpp:72:37: error: 'class IntStack' has no member named 'monotonicIncreasing'
   72 |     std::vector<int> result = stack.monotonicIncreasing();
      |                                     ^~~~~~~~~~~~~~~~~~~
Compilation failed for Task 304048
----------------------
Compiling and running base_code.cpp for Task 307570...
[==========] Running 5 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 5 tests from DataPipelineTest
[ RUN      ] DataPipelineTest.SingleThreadedBasic
[       OK ] DataPipelineTest.SingleThreadedBasic (0 ms)
[ RUN      ] DataPipelineTest.EnableParallelismTest
tasks/307570/test.cpp:48: Failure
Value of: pipeline.isParallel()
  Actual: false
Expected: true

tasks/307570/test.cpp:49: Failure
Expected equality of these values:
  pipeline.getConcurrencyLevel()
    Which is: 1
  3u
    Which is: 3

[  FAILED  ] DataPipelineTest.EnableParallelismTest (0 ms)
[ RUN      ] DataPipelineTest.ErrorHandling
unknown file: Failure
C++ exception with description "Triggered error in stage!" thrown in the test body.

[  FAILED  ] DataPipelineTest.ErrorHandling (0 ms)
[ RUN      ] DataPipelineTest.DrainTest
[       OK ] DataPipelineTest.DrainTest (0 ms)
[ RUN      ] DataPipelineTest.PushAfterStop
[       OK ] DataPipelineTest.PushAfterStop (0 ms)
[----------] 5 tests from DataPipelineTest (0 ms total)

[----------] Global test environment tear-down
[==========] 5 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] DataPipelineTest.EnableParallelismTest
[  FAILED  ] DataPipelineTest.ErrorHandling

 2 FAILED TESTS
Completed Task 307570
----------------------
Compiling and running base_code.cpp for Task 307623...
In file included from tasks/307623/test.cpp:1:
tasks/307623/base_code.cpp:16:5: error: 'Priority' does not name a type
   16 |     Priority priority;
      |     ^~~~~~~~
tasks/307623/base_code.cpp: In member function 'bool Task::operator<(const Task&) const':
tasks/307623/base_code.cpp:22:13: error: 'priority' was not declared in this scope; did you mean 'setpriority'?
   22 |         if (priority != other.priority)
      |             ^~~~~~~~
      |             setpriority
tasks/307623/base_code.cpp:22:31: error: 'const struct Task' has no member named 'priority'
   22 |         if (priority != other.priority)
      |                               ^~~~~~~~
tasks/307623/base_code.cpp:23:37: error: 'const struct Task' has no member named 'priority'
   23 |             return priority > other.priority;
      |                                     ^~~~~~~~
tasks/307623/base_code.cpp: At global scope:
tasks/307623/base_code.cpp:79:10: error: extra qualification 'ThreadPool::' on member 'submit' [-fpermissive]
   79 |     auto ThreadPool::submit(F&& f, Args&&... args)
      |          ^~~~~~~~~~
tasks/307623/base_code.cpp: In member function 'size_t ThreadPool::getTaskQueueSize() const':
tasks/307623/base_code.cpp:105:5: warning: no return statement in function returning non-void [-Wreturn-type]
  105 |     }
      |     ^
tasks/307623/base_code.cpp: In member function 'size_t ThreadPool::getActiveThreadCount() const':
tasks/307623/base_code.cpp:109:5: warning: no return statement in function returning non-void [-Wreturn-type]
  109 |     }
      |     ^
tasks/307623/base_code.cpp: In member function 'size_t ThreadPool::getTotalTasksExecuted() const':
tasks/307623/base_code.cpp:113:5: warning: no return statement in function returning non-void [-Wreturn-type]
  113 |     }
      |     ^
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_BasicTaskExecution_Test::TestBody()':
tasks/307623/test.cpp:33:30: error: 'Priority' has not been declared
   33 |     auto fut1 = pool->submit(Priority::Medium, simulateWork, 100, std::ref(counter));
      |                              ^~~~~~~~
tasks/307623/test.cpp:34:30: error: 'Priority' has not been declared
   34 |     auto fut2 = pool->submit(Priority::High, simulateWork, 100, std::ref(counter));
      |                              ^~~~~~~~
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_DynamicResizingIncrease_Test::TestBody()':
tasks/307623/test.cpp:54:22: error: 'Priority' has not been declared
   54 |         pool->submit(Priority::Medium, task);
      |                      ^~~~~~~~
tasks/307623/test.cpp:58:11: error: 'class ThreadPool' has no member named 'resize'
   58 |     pool->resize(5);
      |           ^~~~~~
tasks/307623/test.cpp:62:22: error: 'Priority' has not been declared
   62 |         pool->submit(Priority::Low, task);
      |                      ^~~~~~~~
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_TaskDependencies_Test::TestBody()':
tasks/307623/test.cpp:95:30: error: 'Priority' has not been declared
   95 |     auto fut1 = pool->submit(Priority::Low, task1);
      |                              ^~~~~~~~
tasks/307623/test.cpp:96:30: error: 'Priority' has not been declared
   96 |     auto fut2 = pool->submit(Priority::Low, task2);
      |                              ^~~~~~~~
tasks/307623/test.cpp:102:30: error: 'Priority' has not been declared
  102 |     auto fut3 = pool->submit(Priority::High, std::move(dependencies), task3);
      |                              ^~~~~~~~
In file included from /usr/local/include/gtest/gtest-printers.h:122,
                 from /usr/local/include/gtest/gtest-matchers.h:49,
                 from /usr/local/include/gtest/internal/gtest-death-test-internal.h:47,
                 from /usr/local/include/gtest/gtest-death-test.h:43,
                 from /usr/local/include/gtest/gtest.h:64,
                 from tasks/307623/test.cpp:2:
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_SubmitAfterShutdown_Test::TestBody()':
tasks/307623/test.cpp:116:22: error: 'Priority' has not been declared
  116 |         pool->submit(Priority::Medium, task),
      |                      ^~~~~~~~
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_CircularDependencies_Test::TestBody()':
tasks/307623/test.cpp:134:30: error: 'Priority' has not been declared
  134 |     auto fut1 = pool->submit(Priority::Medium, task1);
      |                              ^~~~~~~~
tasks/307623/test.cpp:139:30: error: 'Priority' has not been declared
  139 |     auto fut2 = pool->submit(Priority::Medium, std::move(dependencies1), task2);
      |                              ^~~~~~~~
tasks/307623/test.cpp:146:30: error: 'Priority' has not been declared
  146 |     auto fut3 = pool->submit(Priority::Medium, std::move(dependencies2), task1);
      |                              ^~~~~~~~
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_ResizeToZero_Test::TestBody()':
tasks/307623/test.cpp:162:30: error: 'Priority' has not been declared
  162 |     auto fut1 = pool->submit(Priority::Medium, task);
      |                              ^~~~~~~~
tasks/307623/test.cpp:163:30: error: 'Priority' has not been declared
  163 |     auto fut2 = pool->submit(Priority::Medium, task);
      |                              ^~~~~~~~
tasks/307623/test.cpp:166:11: error: 'class ThreadPool' has no member named 'resize'
  166 |     pool->resize(0);
      |           ^~~~~~
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_ComplexTaskDependencies_Test::TestBody()':
tasks/307623/test.cpp:208:30: error: 'Priority' has not been declared
  208 |     auto futA = pool->submit(Priority::High, taskA);
      |                              ^~~~~~~~
tasks/307623/test.cpp:209:30: error: 'Priority' has not been declared
  209 |     auto futB = pool->submit(Priority::High, taskB);
      |                              ^~~~~~~~
tasks/307623/test.cpp:215:30: error: 'Priority' has not been declared
  215 |     auto futC = pool->submit(Priority::Medium, std::move(dependencies1), taskC);
      |                              ^~~~~~~~
tasks/307623/test.cpp:220:30: error: 'Priority' has not been declared
  220 |     auto futD = pool->submit(Priority::Low, std::move(dependencies2), taskD);
      |                              ^~~~~~~~
tasks/307623/test.cpp: In member function 'virtual void ThreadPoolTest_HighConcurrencySubmissions_Test::TestBody()':
tasks/307623/test.cpp:246:46: error: 'Priority' has not been declared
  246 |         futures.emplace_back(highPool.submit(Priority::Medium, task));
      |                                              ^~~~~~~~
Compilation failed for Task 307623
----------------------
Compiling and running base_code.cpp for Task 473609...
Running main() from /Users/ittehad/Documents/Coding/Python/Turing/Tester_cpp_codebench/tasks/303997/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 6 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 6 tests from DijkstraTest
[ RUN      ] DijkstraTest.PositiveNoNegativeEdges
[       OK ] DijkstraTest.PositiveNoNegativeEdges (0 ms)
[ RUN      ] DijkstraTest.NegativeEdgeWeightsNoCycle
[       OK ] DijkstraTest.NegativeEdgeWeightsNoCycle (0 ms)
[ RUN      ] DijkstraTest.NegativeCycleDetection
tasks/473609/test.cpp:57: Failure
Expected equality of these values:
  distances[0]
    Which is: -5
  0

tasks/473609/test.cpp:58: Failure
Expected equality of these values:
  distances[1]
    Which is: 2
  0

tasks/473609/test.cpp:59: Failure
Expected equality of these values:
  distances[2]
    Which is: -1
  0

[  FAILED  ] DijkstraTest.NegativeCycleDetection (0 ms)
[ RUN      ] DijkstraTest.UnreachableNodes
[       OK ] DijkstraTest.UnreachableNodes (0 ms)
[ RUN      ] DijkstraTest.EmptyGraph
[       OK ] DijkstraTest.EmptyGraph (0 ms)
[ RUN      ] DijkstraTest.WrongInput
[       OK ] DijkstraTest.WrongInput (0 ms)
[----------] 6 tests from DijkstraTest (0 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 5 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] DijkstraTest.NegativeCycleDetection

 1 FAILED TEST
Completed Task 473609
----------------------
Compiling and running base_code.cpp for Task 473631...
Running main() from /Users/ittehad/Documents/Coding/Python/Turing/Tester_cpp_codebench/tasks/303997/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from SolutionTest
[ RUN      ] SolutionTest.test1
[       OK ] SolutionTest.test1 (0 ms)
[ RUN      ] SolutionTest.test2
[       OK ] SolutionTest.test2 (0 ms)
[ RUN      ] SolutionTest.test3
[       OK ] SolutionTest.test3 (0 ms)
[ RUN      ] SolutionTest.test4
[       OK ] SolutionTest.test4 (0 ms)
[----------] 4 tests from SolutionTest (0 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 4 tests.
Completed Task 473631
----------------------
Compiling and running base_code.cpp for Task 473646...
Running main() from /Users/ittehad/Documents/Coding/Python/Turing/Tester_cpp_codebench/tasks/303997/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 5 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 5 tests from ParseEthernetPacketTest
[ RUN      ] ParseEthernetPacketTest.ValidPacketTest
tasks/473646/test.cpp:31: Failure
Expected equality of these values:
  parsedFields["CRC Validation"]
    Which is: ""
  "Valid"

[  FAILED  ] ParseEthernetPacketTest.ValidPacketTest (0 ms)
[ RUN      ] ParseEthernetPacketTest.InvalidCRC
tasks/473646/test.cpp:54: Failure
Expected equality of these values:
  parsedFields["CRC Validation"]
    Which is: ""
  "Invalid"

[  FAILED  ] ParseEthernetPacketTest.InvalidCRC (0 ms)
[ RUN      ] ParseEthernetPacketTest.EmptyPacket
[       OK ] ParseEthernetPacketTest.EmptyPacket (0 ms)
[ RUN      ] ParseEthernetPacketTest.UnsupportedEtherType
[       OK ] ParseEthernetPacketTest.UnsupportedEtherType (0 ms)
[ RUN      ] ParseEthernetPacketTest.ShortPacket
[       OK ] ParseEthernetPacketTest.ShortPacket (0 ms)
[----------] 5 tests from ParseEthernetPacketTest (0 ms total)

[----------] Global test environment tear-down
[==========] 5 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] ParseEthernetPacketTest.ValidPacketTest
[  FAILED  ] ParseEthernetPacketTest.InvalidCRC

 2 FAILED TESTS
Completed Task 473646
----------------------
Compiling and running base_code.cpp for Task 473652...
tasks/473652/test.cpp: In member function 'virtual void ElGamalTest_EncryptionDecryption_Test::TestBody()':
tasks/473652/test.cpp:5:5: error: 'ElGamal' was not declared in this scope
    5 |     ElGamal elgamal;
      |     ^~~~~~~
tasks/473652/test.cpp:7:29: error: 'elgamal' was not declared in this scope; did you mean 'lgammal'?
    7 |     auto encryptedMessage = elgamal.Encrypt(message);
      |                             ^~~~~~~
      |                             lgammal
tasks/473652/test.cpp: In member function 'virtual void ElGamalTest_EmptyMessage_Test::TestBody()':
tasks/473652/test.cpp:13:5: error: 'ElGamal' was not declared in this scope
   13 |     ElGamal elgamal;
      |     ^~~~~~~
tasks/473652/test.cpp:15:29: error: 'elgamal' was not declared in this scope; did you mean 'lgammal'?
   15 |     auto encryptedMessage = elgamal.Encrypt(message);
      |                             ^~~~~~~
      |                             lgammal
tasks/473652/test.cpp: In member function 'virtual void ElGamalTest_SingleCharacter_Test::TestBody()':
tasks/473652/test.cpp:21:5: error: 'ElGamal' was not declared in this scope
   21 |     ElGamal elgamal;
      |     ^~~~~~~
tasks/473652/test.cpp:23:29: error: 'elgamal' was not declared in this scope; did you mean 'lgammal'?
   23 |     auto encryptedMessage = elgamal.Encrypt(message);
      |                             ^~~~~~~
      |                             lgammal
tasks/473652/test.cpp: In member function 'virtual void ElGamalTest_SpecialCharacters_Test::TestBody()':
tasks/473652/test.cpp:29:5: error: 'ElGamal' was not declared in this scope
   29 |     ElGamal elgamal;
      |     ^~~~~~~
tasks/473652/test.cpp:31:29: error: 'elgamal' was not declared in this scope; did you mean 'lgammal'?
   31 |     auto encryptedMessage = elgamal.Encrypt(message);
      |                             ^~~~~~~
      |                             lgammal
tasks/473652/test.cpp: In member function 'virtual void ElGamalTest_LongMessage_Test::TestBody()':
tasks/473652/test.cpp:37:5: error: 'ElGamal' was not declared in this scope
   37 |     ElGamal elgamal;
      |     ^~~~~~~
tasks/473652/test.cpp:39:29: error: 'elgamal' was not declared in this scope; did you mean 'lgammal'?
   39 |     auto encryptedMessage = elgamal.Encrypt(message);
      |                             ^~~~~~~
      |                             lgammal
tasks/473652/test.cpp: In member function 'virtual void ElGamalTest_EdgeCharacters_Test::TestBody()':
tasks/473652/test.cpp:45:5: error: 'ElGamal' was not declared in this scope
   45 |     ElGamal elgamal;
      |     ^~~~~~~
tasks/473652/test.cpp:47:29: error: 'elgamal' was not declared in this scope; did you mean 'lgammal'?
   47 |     auto encryptedMessage = elgamal.Encrypt(message);
      |                             ^~~~~~~
      |                             lgammal
tasks/473652/test.cpp: At global scope:
tasks/473652/test.cpp:52:5: error: conflicting declaration of C function 'int main(int, char**)'
   52 | int main(int argc, char **argv) {
      |     ^~~~
In file included from tasks/473652/test.cpp:1:
tasks/473652/base_code.cpp:21:5: note: previous declaration 'int main()'
   21 | int main() {
      |     ^~~~
Compilation failed for Task 473652
----------------------
Compiling and running base_code.cpp for Task 473660...
In file included from /usr/local/include/gtest/gtest-printers.h:122,
                 from /usr/local/include/gtest/gtest-matchers.h:49,
                 from /usr/local/include/gtest/internal/gtest-death-test-internal.h:47,
                 from /usr/local/include/gtest/gtest-death-test.h:43,
                 from /usr/local/include/gtest/gtest.h:64,
                 from tasks/473660/test.cpp:1:
tasks/473660/test.cpp:10:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_SimpleXMLToJsonConversion_Test::test_info_'
   10 | TEST(XMLToJSONConverterTest, SimpleXMLToJsonConversion) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:10:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_SimpleXMLToJsonConversion_Test::TestBody' within 'XMLToJSONConverter'
   10 | TEST(XMLToJSONConverterTest, SimpleXMLToJsonConversion) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:33:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_AddNullChild_Test::test_info_'
   33 | TEST(XMLToJSONConverterTest, AddNullChild) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:33:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_AddNullChild_Test::TestBody' within 'XMLToJSONConverter'
   33 | TEST(XMLToJSONConverterTest, AddNullChild) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:42:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_NullRootNode_Test::test_info_'
   42 | TEST(XMLToJSONConverterTest, NullRootNode) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:42:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_NullRootNode_Test::TestBody' within 'XMLToJSONConverter'
   42 | TEST(XMLToJSONConverterTest, NullRootNode) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:49:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_CircularReferenceDetection_Test::test_info_'
   49 | TEST(XMLToJSONConverterTest, CircularReferenceDetection) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:49:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_CircularReferenceDetection_Test::TestBody' within 'XMLToJSONConverter'
   49 | TEST(XMLToJSONConverterTest, CircularReferenceDetection) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:63:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_EmptyNode_Test::test_info_'
   63 | TEST(XMLToJSONConverterTest, EmptyNode) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:63:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_EmptyNode_Test::TestBody' within 'XMLToJSONConverter'
   63 | TEST(XMLToJSONConverterTest, EmptyNode) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:74:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_SingleNodeWithValue_Test::test_info_'
   74 | TEST(XMLToJSONConverterTest, SingleNodeWithValue) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:74:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_SingleNodeWithValue_Test::TestBody' within 'XMLToJSONConverter'
   74 | TEST(XMLToJSONConverterTest, SingleNodeWithValue) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:85:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_SpecialCharactersInNode_Test::test_info_'
   85 | TEST(XMLToJSONConverterTest, SpecialCharactersInNode) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:85:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_SpecialCharactersInNode_Test::TestBody' within 'XMLToJSONConverter'
   85 | TEST(XMLToJSONConverterTest, SpecialCharactersInNode) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:103:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_LargeTree_Test::test_info_'
  103 | TEST(XMLToJSONConverterTest, LargeTree) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:103:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_LargeTree_Test::TestBody' within 'XMLToJSONConverter'
  103 | TEST(XMLToJSONConverterTest, LargeTree) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:122:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_NestedXMLWithChildren_Test::test_info_'
  122 | TEST(XMLToJSONConverterTest, NestedXMLWithChildren) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:122:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_NestedXMLWithChildren_Test::TestBody' within 'XMLToJSONConverter'
  122 | TEST(XMLToJSONConverterTest, NestedXMLWithChildren) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:151:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_NodeWithEmptyNameAndValue_Test::test_info_'
  151 | TEST(XMLToJSONConverterTest, NodeWithEmptyNameAndValue) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:151:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_NodeWithEmptyNameAndValue_Test::TestBody' within 'XMLToJSONConverter'
  151 | TEST(XMLToJSONConverterTest, NodeWithEmptyNameAndValue) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:162:6: error: invalid use of qualified-name 'XMLToJSONConverter::XMLToJSONConverterTest_SpecialCharacterInValue_Test::test_info_'
  162 | TEST(XMLToJSONConverterTest, SpecialCharacterInValue) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:162:6: error: cannot define member function 'XMLToJSONConverter::XMLToJSONConverterTest_SpecialCharacterInValue_Test::TestBody' within 'XMLToJSONConverter'
  162 | TEST(XMLToJSONConverterTest, SpecialCharacterInValue) {
      |      ^~~~~~~~~~~~~~~~~~~~~~
tasks/473660/test.cpp:170:2: error: expected '}' at end of input
  170 | }
      |  ^
In file included from tasks/473660/test.cpp:5:
tasks/473660/base_code.cpp:19:26: note: to match this '{'
   19 | class XMLToJSONConverter {
      |                          ^
tasks/473660/base_code.cpp: In static member function 'static std::string XMLToJSONConverter::convert(XMLNode*)':
tasks/473660/base_code.cpp:24:1: warning: no return statement in function returning non-void [-Wreturn-type]
   24 | };
      | ^
tasks/473660/test.cpp: At global scope:
tasks/473660/test.cpp:170:2: error: expected unqualified-id at end of input
  170 | }
      |  ^
Compilation failed for Task 473660
----------------------
